### ISSUE TITLE
Use module such as 'fs' within a script?

### ISSUE THREAD

**Issue by dandavison** [AnWeber/vscode-statusbar-command#21]
I'd like to create a statusbar command that displays some text returned by a function in the node `fs` module. Specifically, I'm trying:

```
"script": "statusBarItem.text = require('fs').readlinkSync('/tmp/mylink.txt')"
```

However, from experimenting, I believe that `fs` is not available in the script context and that `require('fs')` doesn't work (although I failed to catch the error when I tried using `try...catch` in the script).

Is it possible to do what I'm trying to do here, or what would it take to make it possible?

**Comment by AnWeber** [comment]
I currently use `vm.runInNewContext`. Here no access to the [local scope or require is possible](https://nodejs.org/api/vm.html#scriptruninnewcontextcontextobject-options). I change it, but only within Trusted Workspaces.

**Comment by dandavison** [comment]
Fantastic, thanks again for the quick and very helpful response.

**Comment by AnWeber** [comment]
I have integrated an OutputChannel that would output errors in the script area. Did you find the channel
![outputchannel](https://user-images.githubusercontent.com/13434739/186509721-971acf87-8903-447d-a25f-9f9ffbed7fa2.gif)

Also, it would be possible to debug that script. For this I have created a [howto for vscode-httpyac](https://github.com/AnWeber/vscode-httpyac/issues/13#issuecomment-780646242). The principle is the same.
.

**Comment by dandavison** [comment]
Hi @AnWeber, I don't _think_ this is working for me. I've installed from a `.vsix` built on master c7ed250a71ccfaa1f66f2613f0fa9801a16c7712, and I'm doing the following experiments:

```
"script": "console.log(vscode.workspace.isTrusted); statusBarItem.text = 'hello'"
```

I see `true` in the developer tools JS console, and `hello` appears in the status bar.

```
"script": "console.log(vscode.workspace.isTrusted); require('fs'); statusBarItem.text = 'hello'",
```

I still see `true` in the dev tools console, but the status bar text is now whatever I have set in the `"text"` field of the `"statusbar_command.commands"` config entry.

**Comment by dandavison** [comment]
The output channel is emitting lines that look like "empty JS object" syntax. Thanks for your debugging tips above; I haven't looked into any of this properly yet! I'll try to follow your suggestions above when I get a bit more time.

<img width="650" alt="image" src="https://user-images.githubusercontent.com/52205/186681121-27a37b3d-dc3a-4013-a669-3a578262ee9e.png">

**Comment by AnWeber** [comment]
@dandavison  There were still a few small errors in the code. I have now created a test case (read package version) and tested it.

- output of `{}` was an error because of the execution with `runInContext`. Error Class is encapsulated in some kind of proxy and therefore `instanceOf Error` does not work correctly 
- `require` is actually not directly available. I have now included the simple variant, but it resolves the `require` in the context of my extension. Meaning own dependencies in the current workspace are not available.
- Another error could be the execution directory of the script. This is not the current workspace but the location of my extension. For this reason absolute paths should be used for referencing files (``${workspaceFolder.fsPath}/package.json``) . 

```json
{
      "alignment": "left",
      "command": "workbench.action.terminal.toggleTerminal",
      "id": "sbc.packageVersion",
      "text": "Package Version",
      "scriptEvents": ["vscode.workspace.onDidChangeConfiguration", "vscode.window.onDidChangeActiveTextEditor"],
      "script": "workspaceFolder && (statusBarItem.text = (JSON.parse(require('fs').readFileSync(`${workspaceFolder.fsPath}/package.json`, 'utf8'))).version)"
    }
```
